R/psis_approximate_posterior.R

```diff
@@ -0,0 +1,46 @@
+#' Diagnostics for Laplace and ADVI approximations and Laplace-loo and ADVI-loo
+#'
+#' @param log_p The log-posterior (target) evaluated at samples from the proposal distribution (q).
+#' @param log_q The log-density (proposal) evaluated at samples from the proposal distribution (q).
+#' @param log_liks A log-likelihood matrix. See \code{\link{loo.matrix}}. Default is \code{NULL}. Then only the posterior is evaluated using the k_hat diagnostic.
+#' @inheritParams loo
+#'
+#' @return
+#' If log likelihoods are supplied, the function returns a \code{loo} object,
+#' otherwise the function returns a \code{psis} object.
+#'
+#' @seealso \code{loo} and \code{psis}
+#'
+#' @template loo-and-psis-references
+#'
+#' @export
+psis_approximate_posterior <- function(log_p, log_q, log_liks = NULL, cores, save_psis, ...){
```

Comment:
I think there should be explicit defaults for the `cores` and `save_psis` arguments, e.g.,

```suggestion
psis_approximate_posterior <- function(log_p, log_q, log_liks = NULL, cores = getOption("mc.cores", 1), save_psis = FALSE, ...){
```

Reply:
Sure! I assumed since the function will not be a part of the public API I assume you did not want any defaults. In my experience, non-API functions using defaults can miss usage bugs in accessing the function, and hence not using defaults will be more defensive.  But you know how you want it so I do what you think is best and is in line with the overall structure of the project. Just let me know how you want it.

Reply:
Oh, I was thinking this would maybe be exposed to users, but if it won't be part of the public API then I agree. It's fine to leave it without defaults.

Reply:
Yes. Already fixed.

Reply:
If I remember correctly we said that I should implement it as a standalone function and then you could plug it in with the laplace objects since there are currently just a list.